agimus_demos_common/launch/franka_common.launch.py

```diff
@@ -122,13 +123,59 @@ def launch_setup(
         value_type=str,
     )
 
+    xacro_collision_args = {
+        "robot_ip": robot_ip,
+        "arm_id": arm_id,
+        "ros2_control": "true",
+        "hand": "true",
+        "use_fake_hardware": "false",
+        "fake_sensor_commands": "false",
+        "gazebo": "false",
+        "ee_id": "franka_hand",
+        "gazebo_effort": "true",
+        "with_sc": "true",
+        "franka_controllers_params": franka_controllers_params,
+    }
+    robot_collision_description = ParameterValue(
+        Command(
+            [
+                PathJoinSubstitution([FindExecutable(name="xacro")]),
+                " ",
+                PathJoinSubstitution(
+                    [
+                        FindPackageShare("franka_description"),
+                        "robots",
+                        arm_id_str,
+                        f"{arm_id_str}.urdf.xacro",
+                    ]
+                ),
+                # Convert dict to list of parameters
+                *[
+                    arg
+                    for key, val in xacro_collision_args.items()
+                    for arg in (f" {key}:=", val)
+                ],
+            ]
+        ),
+        value_type=str,
+    )
+
     robot_state_publisher_node = Node(
         package="robot_state_publisher",
         executable="robot_state_publisher",
         parameters=[get_use_sime_time(), {"robot_description": robot_description}],
         output="screen",
     )
 
+    robot_collision_publisher_node = Node(
+        package="robot_state_publisher",
+        executable="robot_state_publisher",
+        name="robot_collision_publisher",
+        output="screen",
+        remappings=[("robot_description", "robot_description_with_collision")],
```

Comment:
```suggestion
        remappings=[("robot_description", "robot_description_with_collision"), ("/tf", "/_null/tf"), ("/tf_static", "/_null/tf_static")],
```
We want transformations to only come from the default `robot_state_publisher` and not this node

Reply:
I don't get why we need a `robot_state_publisher` if its output are to be ignored :thinking:

Reply:
actually I was wondering if we simply should add a robot_description to the agimus_controller node directly?
Instead of publishing this model here?

Reply:
@MaximilienNaveau suggestion makes sense to me.

Reply:
It shouldn't be a `robot_state_publisher` node.